client/src/pages/PublicProfile.js

```diff
@@ -0,0 +1,72 @@
+import { useState, useContext } from 'react';
+import styled from 'styled-components';
+import useStyles from '../themes/theme.profile';
+import { useHistory } from 'react-router-dom';
+
+import AppBar from '@material-ui/core/AppBar';
+
+import Navbar from '../components/Navbar';
+import Sidebar from '../components/Sidebar/index';
+import CreatePoll from '../components/CreatePoll/index';
+import Avatar from '@material-ui/core/Avatar';
+
+import UserContext from '../context/UserContext';
+ 
+export default function PublicProfile () {   
+    const classes = useStyles();
+    const { userData } = useContext(UserContext);
+    const [openCreatePoll, setOpenCreatePoll] = useState(false);
+    const history = useHistory();
+    const { user } = history.location.state;
```

Comment:
what happens if you visit this page on it's own for example /1234/profile route, i hit refresh, would by history location state still exist?

Reply:
Good call. Does it make more sense the way I have refactored it below?